app/views/planning_applications/_assess_proposal.html.erb

```diff
@@ -1,96 +1,62 @@
-          <% if @planning_application.determined? %>
-            <%= tag.strong "Completed", id: "#{step_name.parameterize}-completed", class: "govuk-tag app-task-list__task-completed" %>
-          <% end %>
-        </li>
-      </ul>
-    </li>
-  <% end %>
+  <li>
+    <h2 class="app-task-list__section">
+      <span class="app-task-list__section-number"><%= t("#{@planning_application.status}.heading") %></span>
+    </h2>
+    <ul class="app-task-list__items">
+      <li class="app-task-list__item">
+        <% step_name = t("#{@planning_application.status}.proposal_step") %>
+
+        <% if (current_user.assessor? || current_user.admin?) %>
+          <% aria_attributes = @planning_application.assessor_decision ? { describedby: "#{step_name.parameterize}-completed" } : {} %>
+
+          <%# TODO: Make a helper for this %>
+          <%
+            path = @planning_application.assessor_decision ?
+              edit_planning_application_decision_path(@planning_application) :
+                new_planning_application_decision_path(@planning_application)
+          %>
+
+          <span class="app-task-list__task-name"><%= link_to step_name, path, aria: aria_attributes %></span>
+        <% elsif !@planning_application.in_assessment? && (current_user.reviewer? || current_user.admin?) %>
+          <% aria_attributes = @planning_application.reviewer_decision ? { describedby: "#{step_name.parameterize}-completed" } : {} %>
+
+          <%# TODO: Make a helper for this %>
+          <%
+            path = @planning_application.reviewer_decision ?
+              edit_planning_application_decision_path(@planning_application) :
+                new_planning_application_decision_path(@planning_application)
+          %>
+
+          <span class="app-task-list__task-name"><%= link_to step_name, path, aria: aria_attributes %></span>
+        <% else %>
+          <span class="app-task-list__task-name"><%= step_name %></span>
+        <% end %>
+
+        <% if proposal_step_mark_completed?(step_name, @planning_application) %>
+          <%= tag.strong "Completed", id: "#{step_name.parameterize}-completed", class: "govuk-tag app-task-list__task-completed" %>
+        <% end %>
+      </li>
+
+      <li class="app-task-list__item">
+        <% step_name = t("#{@planning_application.status}.recommendation_step") %>
+
+        <% if (current_user.assessor? || current_user.admin?) && @planning_application.assessor_decision %>
+          <% aria_attributes = @planning_application.awaiting_determination? ? { describedby: "#{step_name.parameterize}-completed" } : {} %>
+
+          <span class="app-task-list__task-name"><%= link_to step_name, edit_planning_application_path(@planning_application), aria: aria_attributes %></span>
+        <% elsif current_user.reviewer? && @planning_application.reviewer_decision %>
+          <% aria_attributes = @planning_application.awaiting_determination? ? { describedby: "#{step_name.parameterize}-completed" } : {} %>
+
```

Comment:
It looks like we can reduce some duplication by setting `aria_attributes` before the `if`